### DISCUSSION TITLE
Can I use AWS RDS database URL instead of Supabase URL in hosted Supabase?

### DISCUSSION THREAD

**Discussion by sakkurthi-sashank** [supabase/supabase#21081] [Questions]
I am currently working on a project that utilizes Supabase as the backend for my application. However, due to certain requirements and constraints, I am exploring the possibility of using an AWS RDS database instead of Supabase's native solution.

**Comment by GaryAustin1** [comment]
You can't on Supabase hosted.
You might be able to self-hosted.
https://supabase.com/docs/guides/self-hosting
https://supabase.com/docs/guides/realtime/bring-your-own-database
https://github.com/supabase-community/supabase-on-aws

**Reply by rantoniuk** [reply]
@GaryAustin1 it seems that the "Bring your own database" guide disappeared from the docs. Is there any recommended way to use an external (AWS RDS PGSQL) instance instead of the bundled one?
Let's even assume I don't need to use realtime, but only e.g. expose REST/GQL.

**Comment by johndpope** [comment]
actually the lost link has rds instructions - 
https://web.archive.org/web/20240321083806/https://supabase.com/docs/guides/realtime/bring-your-own-database

I started pulling apart the containers and stuff to boot up on rds
i have a test script for realtime  - and adjust the  schemas for rds permisisons - users 

i spent a lot of time on this - still not there - it's a work in progress
https://github.com/johndpope/rds-supabase/blob/main/docker/bootstrap_db.sh

https://github.com/johndpope/rds-supabase/tree/main/docker

related - https://github.com/orgs/supabase/discussions/21081

**Reply by raffidahmad** [reply]
have you been able to succeed? I'm trying to do the same thing

**Reply by johndpope** [reply]
i haven't been able to get real time working - 

this is customized to aws - rds can't handle some extensions. eg. pg_net
https://github.com/johndpope/rds-supabase/blob/main/docker/rds_schema.sql

my boss prebought rds instance - and i sunk > 1 month investigating. 
the auth container / realtime boot up connected to rds 
i bypass migrations - because sql /elixir code is not compatible with some things
https://github.com/supabase/realtime/issues?q=is%3Aissue%20state%3Aopen%20rds

make sure you edit this - 
```sql
ALTER ROLE supabase_admin SET search_path TO auth, public;
```

i basically go to this point. but had to park things as it burning up too much time.


```shell
#!/bin/bash

# Set tenant variables
TENANT_NAME="realtime-dev2"
TENANT_EXTERNAL_ID="realtime-dev2"

DB_PORT="5432"
REGION="ap-southeast-2"
POSTGRES_CDC_DEFAULT="postgres_cdc_rls"

# Use the correct PostgreSQL endpoint and database
DB_HOST="blabla.ap-southeast-2.rds.amazonaws.com"
DB_NAME=""
DB_USER=""
DB_PASS=""

# Get the JWT token - MUST have postgres role
node ./node-debug/generate-token.js > admin_token.txt
JWT_TOKEN=$(cat admin_token.txt)

# Realtime base URL
REALTIME_BASE="http://0.0.0.0:8000/realtime/v1/api"

# Use your service role key for authentication
SERVICE_ROLE_KEY="blabla"

echo "Registering tenant: $TENANT_NAME (external_id: $TENANT_EXTERNAL_ID)"
echo "Using database: $DB_NAME on $DB_HOST"

# Make the API request with corrected database settings
# curl -X POST \
#   -H "Content-Type: application/json" \
#   -H "Authorization: Bearer ${JWT_TOKEN}" \
#   -H "apikey: ${SERVICE_ROLE_KEY}" \
#   -d "{
#     \"tenant\": {
#       \"name\": \"$TENANT_NAME\",
#       \"external_id\": \"$TENANT_EXTERNAL_ID\",
#       \"jwt_secret\": \"your-super-secret-jwt-token-with-at-least-32-characters-long\",
#       \"postgres_cdc_default\": \"$POSTGRES_CDC_DEFAULT\",
#       \"extensions\": [{
#         \"type\": \"postgres_cdc_rls\",
#         \"settings\": {
#           \"db_name\": \"$DB_NAME\",
#           \"db_host\": \"$DB_HOST\",
#           \"db_user\": \"$DB_USER\",
#           \"db_password\": \"$DB_PASS\",
#           \"db_port\": \"$DB_PORT\",
#           \"region\": \"$REGION\",
#           \"poll_interval_ms\": 100,
#           \"poll_max_record_bytes\": 1048576
#         }
#       }]
#     }
#   }" \
#   "${REALTIME_BASE}/tenants"


nc -zv ${DB_HOST} 5432

# Health check
curl -X GET \
  -H "Authorization: Bearer ${JWT_TOKEN}" \
  -H "apikey: ${SERVICE_ROLE_KEY}" \
  "${REALTIME_BASE}/tenants/${TENANT_EXTERNAL_ID}/health"


```


generate-token.js
```javascript
// Save as generate-token.js
const jwt = require('jsonwebtoken');

const secret = 'your-super-secret-jwt-token-with-at-least-32-characters-long'; // Replace with your actual API_JWT_SECRET
const payload = {
  iss: 'supabase',
  iat: Math.floor(Date.now() / 1000),
  exp: Math.floor(Date.now() / 1000) + (365 * 24 * 60 * 60), // 1 year expiry
  aud: 'realtime',
  sub: 'admin',
  role: 'postgres'
};

const token = jwt.sign(payload, secret);
console.log(token);

// npm install jsonwebtoken
// node generate-token.js



```

some of the containers can get a stale password with docker compose build - you kinda have to rebuild --nocache each and every time.

**Reply by raffidahmad** [reply]
kind of in the same boat as you, i was thinking to use the supabase-aws cloudformation template to make the RDS only and then host all the other services in a single docker (the template provided seems to deploy all services in separate ECS containers, tho i havent actually run it yet)

Template: https://github.com/supabase-community/supabase-on-aws

**Reply by johndpope** [reply]
i dont think it will wont work completely - its using obsolete library coder / or something... check the issues.
